### PR TITLE
Warn about unsupported operations in DummyCategory and support "list_of_lists_of_morphisms" in CategoryConstructor

### PR DESCRIPTION
**Pull request by zickgraf**

### CODE REVIEW

File: CAP/gap/DummyCategory.gi [hunk 1]
```diff
@@ -77,6 +77,16 @@ InstallMethod( DummyCategory,
     
     Finalize( C );
     
+    for operation_name in options.list_of_operations_to_install do
+        
+        if not CanCompute( C, operation_name ) then
+            
+            Print( "WARNING: The dummy category cannot compute ", operation_name, ", probably because the operation is not supported by CategoryConstructor yet.\n" );
```
[the remark this answers is not in the document]
Comment (zickgraf) [reply CAP/gap/DummyCategory.gi:84]: I'm checking `not CanCompute( C, operation_name )` where `C` is a dummy category which was just constructed. I think "cannot compute" is the most fitting phrase.

"DummyCategory will neglect ...." would make sense for me if I would check which operations are supported before actually creating the category. However, the supported operations are an implementation detail of CategoryConstructor, so there is no simple check for this.

[the remark this answers is not in the document]
Comment (zickgraf) [reply CAP/gap/DummyCategory.gi:84]: Discussion resolved by mail.

**Comment by mohamed-barakat** [comment]
> @mohamed-barakat This might change some numbers due to "MorphismBetweenDirectSums" etc. now being handled by CategoryConstructor and thus also by ReinterpretationOfCategory.

Will start a test for `CategoricalTowers`.

**Comment by mohamed-barakat** [comment]
@kamalsaleh

The example file `presheaves_of_algebroid.g` in `FunctorCategories` is failing with this error in `HigherHomologicalAlgebra/ComplexesCategories`:

```gap
Error, can only deal with 'integer', 'object', 'morphism', 'list_of_objects', 'list_of_morphisms', 'pair_of_morphisms' at ~/.gap/pkg/HigherHomologicalAlgebra/ComplexesCategories//gap/Categories.gi:210 called from
func( C[i] ) at ~/software/gap/build/gap-4.12.1/lib/coll.gi:663 called from
List( [ 1 .. Length( info.filter_list ) ], function ( j )
      local type;
      type := info.filter_list[j];
      if j = 1 and type = "category" then
          return "presh_cat";
      elif type in [ "integer", "element_of_commutative_ring_of_linear_structure" ] then
          return Concatenation( "i_arg[", String( j ), "]" );
      elif type = "object" then
          return Concatenation( "ObjectDatum( cat, i_arg[", String( j ), "] )[1]" );
      elif type = "morphism" then
          return Concatenation( "MorphismDatum( i_arg[", String( j ), "] )" );
      elif type = "list_of_objects" then
          return Concatenation( "List( i_arg[", String( j ), "], o -> ObjectDatum( cat, o )[1] )" );
      elif type = "list_of_morphisms" then
          return Concatenation( "List( i_arg[", String( j ), "], m -> MorphismDatum( cat, m ) )" );
      elif type = "pair_of_morphisms" then
          return Concatenation( "Pair( ", "MorphismDatum( cat, i_arg[", String( j ), "][1] )", ", MorphismDatum( cat, i_arg[", String( j ), "][2] )", " )" );
      else
          Error( "can only deal with 'integer', 'object', 'morphism', 'list_of_objects', 'list_of_morphisms', 'pair_of_morphisms'" );
      fi;
      return;
  end ) at ~/.gap/pkg/HigherHomologicalAlgebra/ComplexesCategories//gap/Categories.gi:213 called from
create_func( name, CC ) at ~/.gap/pkg/CAP_project/CAP//gap/CategoryConstructor.gi:325 called from
CategoryConstructor( rec(
    name := name,
    category_filter := category_filter,
    category_object_filter := category_object_filter,
    category_morphism_filter := category_morphism_filter,
    commutative_ring_of_linear_category := commutative_ring,
    properties := properties,
    is_computable := false,
    object_constructor := object_constructor,
    object_datum := object_datum,
    morphism_constructor := morphism_constructor,
    morphism_datum := morphism_datum,
    underlying_category_getter_string := "UnderlyingCategory",
    list_of_operations_to_install := list_of_operations_to_install,
    supports_empty_limits := supports_empty_limits,
    create_func_bool := create_func_bool,
    create_func_object := create_func_object,
    create_func_morphism := create_func_morphism ) ) at ~/.gap/pkg/HigherHomologicalAlgebra/ComplexesCategories//gap/Categories.gi:236 called from
COMPLEXES_CATEGORY_BY_COCHAINS_AS_TOWER( cat ) at ~/.gap/pkg/HigherHomologicalAlgebra/ComplexesCategories//gap/Categories.gi:256 called from
...  at presheaves_of_algebroid.g:140
```

**Pull request merged**